### PR TITLE
Do not crash if git.exe is not found

### LINKED ISSUE #5562: Crash on launch when git not installed
**Do you want to request a *feature* or report a *bug*?**
This is a bug.
**What is the current behavior?**
capture
This Issue occurs on Launch, under a Fresh/New Start (Moved to new PC that didn't have GitExtensions on it before)
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
1. Launch under First Start (Not Tested if it occurs on Existing Systems)
**What is the expected behavior?**
To Launch Properly
**Environment you encounter the issue:**

- GitExtensions version: 3.0.0.3621 - 3.0.03702
- GIT version: N/A - New System
- OS version: Windows 10 Update 1803
- .NET version: N/A - New System

### PR DESCRIPTION
**Pull request by spdr870**
Fixes #5562

The settings dialog has a lot of logic to find git.exe or git.cmd. It even has a download link and an option to manually select the proper path. This code is never reached. A lot of refactoring is done that makes almost all code depend on git.exe to be set to the correct path. I did not change this existing code. Didn't remove or cleanup anything. Not sure how to handle this, but for now I wanted a solution for the 3.00 release.

I looked at the PR @RussKie did earlier and could not get this to work properly. The code is not tested on hidpi settings.

Changes proposed in this pull request:
- Do not crash when git.exe is not found
- Try to locate git.exe (this is also done in the settings dialog, but this code is never reached)
- Offer a download link and manual locate button
 
Screenshots before and after (if PR changes UI):
Before:
image

After:
image

What did I do to test the code and ensure quality:
- Manual testing

Has been tested on (remove any that don't apply):
- GIT 2.19.1
- Windows 10

### CODE REVIEW

File: GitUI/CommandsDialogs/SettingsDialog/Pages/FormFixGitCommand.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        }
+
+        private void ok_Click(object sender, EventArgs e)
+        {
+            Close();
+        }
+
+        private void download_Click(object sender, EventArgs e)
+        {
+            Process.Start(@"https://git-scm.com/");
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/SettingsDialog/Pages/FormFixGitCommand.cs]: In settings we point to the following:
https://github.com/gitextensions/gitextensions/wiki/Git
Reason is to be able to describe versions that are not working

Should be consistent at least

**Comment by gerhardol** [comment]
Note: This adds more strings to translate

**Comment by RussKie** [comment]
Few things:

I propose we use `PSTaskDialog.cTaskDialog`, in a similar fashion as "remove invalid repos" functionality. 
The top option could be - "_locate git_", the bottom option could be "_git install instructions_" - to open our wiki. And the cancel button, of course, that closes the app.

The dialog text could say - "_Could no locate git.exe. Is it installed?_" (or something similar)

Thoughts?

**Comment by spdr870** [comment]
> Few things:
> 
> I propose we use `PSTaskDialog.cTaskDialog`, in a similar fashion as "remove invalid repos" functionality.
> The top option could be - "_locate git_", the bottom option could be "_git install instructions_" - to open our wiki. And the cancel button, of course, that closes the app.
> 
> The dialog text could say - "_Could not locate git.exe. Is git installed?_" (or something similar)
> 
> Thoughts?

I tried ctaskdialog, but I couldn't get it to work without the application started with a form.
Feel free to add to this pr, I do not have much time until tomorrow.

File: GitUI/CommandsDialogs/SettingsDialog/Pages/FormFixGitCommand.cs (hunk 1 above)
Comment (spdr870) [reply GitUI/CommandsDialogs/SettingsDialog/Pages/FormFixGitCommand.cs]: Updated

**Comment by RussKie** [comment]
We just need to update the app.manifest with:
```
<dependency>
  <dependentAssembly>
    <assemblyIdentity type="win32" name="Microsoft.Windows.Common-Controls" 
        version="6.0.0.0" processorArchitecture="*"
        publicKeyToken="6595b64144ccf1df" language="*" />
  </dependentAssembly>
</dependency>
```

Credits: https://stackoverflow.com/a/1901318/2338036

**Comment by RussKie** [comment]
image

**Comment by RussKie** [comment]
Please have a look, if you're happy - squash/merge

**Comment by RussKie** [comment]
Relates to #5481

**Comment by spdr870** [comment]
> Please have a look, if you're happy - squash/merge

I like the looks of it. I will try to test/review later today.

**Comment by TamaHobbit** [comment]
Today I installed GitExtensions 3.0 RC2, which does seem to include this merge; but I still get the crash.

First, I had git 1.9 and GE 3 gave me the old settings dialog including whether the language is set and the committer email; it said that git 2.19.1 or later needed to be installed. So I installed git 2.19.2 for windows, and now GE 3 gives the crash exactly as in your original message. Commandline usage indicates that "git" (it's in PATH) is indeed the new version now.

**Comment by RussKie** [comment]
No, it is not in RC2. It is in v3.00

**Comment by TamaHobbit** [comment]
Ah; thanks. For anyone else encountering this; under the title of the release, click on "X commits since this release", which brings you to a compare page such as [this](https://github.com/gitextensions/gitextensions/compare/v3.00.00-rc2...master), which lists this merge as not being in RC2.

**Comment by RussKie** [comment]
Fixed issues are marked with milestones
image

**Comment by TamaHobbit** [comment]
The problem in my case turned out to be that I had installed 64-bit Git for windows, but my 32-bit GitExtensions installation requires 32-bit Git for windows to be installed.

**Comment by gerhardol** [comment]
GE works with 64-bit Git

**Pull request merged**